smdebug/core/state_store.py

```diff
@@ -26,7 +27,7 @@ def _get_checkpoint_files_in_dir(self, cp_dir):
         checkpoint_files = []
         for child, _, files in os.walk(cp_dir):
             for file in files:
-                if file != METADATA_FILENAME:
+                if file != METADATA_FILENAME and file != METADATA_FILENAME_S3_UPLOADED:
```

Comment:
this is not a right fix, 
you need to handle exception every where we are trying to access file ( The files can be deleted or modeified by a different process) . Assume that files can be concurrently modified by another process.

Reply:
done